components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx

```diff
@@ -60,18 +69,36 @@ const TokenLists = (props: Props) => {
   // routing
   const history = useHistory()
 
-  // memos
-  const nonFungibleTokenList = React.useMemo(
-    () => filteredAssetList.filter(({ asset }) => asset.isErc721),
-    [filteredAssetList]
-  )
+  // methods
+  const selectedAssetFilter = useSelector(({ wallet }: { wallet: WalletState }) => wallet.selectedAssetFilter)
 
-  const fungibleTokenList = React.useMemo(
-    () => filteredAssetList.filter(({ asset }) => !asset.isErc721),
-    [filteredAssetList]
-  )
+  const { computeFiatAmount } = usePricing(tokenPrices)
+
+  const erc721TokenList = React.useMemo(() => filteredAssetList.filter((token) => token.asset.isErc721), [filteredAssetList])
+
+  const sortedAssetList = React.useMemo(() => {
+    const fungibleTokens = filteredAssetList.filter((asset) => !asset.asset.isErc721)
+    if (selectedAssetFilter.id === 'highToLow') {
+      return fungibleTokens.sort(function (a, b) {
+        const aBalance = a.assetBalance
+        const bBalance = b.assetBalance
+        const bFiatBalance = computeFiatAmount(bBalance, b.asset.symbol, b.asset.decimals)
+        const aFiatBalance = computeFiatAmount(aBalance, a.asset.symbol, a.asset.decimals)
+        return bFiatBalance.toNumber() - aFiatBalance.toNumber()
+      })
+    }
+    if (selectedAssetFilter.id === 'lowToHigh') {
+      return fungibleTokens.sort(function (a, b) {
+        const aBalance = a.assetBalance
+        const bBalance = b.assetBalance
+        const bFiatBalance = computeFiatAmount(bBalance, b.asset.symbol, b.asset.decimals)
+        const aFiatBalance = computeFiatAmount(aBalance, a.asset.symbol, a.asset.decimals)
+        return aFiatBalance.toNumber() - bFiatBalance.toNumber()
+      })
+    }
+    return fungibleTokens
+  }, [filteredAssetList, selectedAssetFilter, computeFiatAmount])
 
-  // methods
   // This filters a list of assets when the user types in search bar
   const onFilterAssets = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
This should be refactored. onFilterAssets should set the search value to state, then a memo should run to filter the list when that value changes

Reply:
Done!